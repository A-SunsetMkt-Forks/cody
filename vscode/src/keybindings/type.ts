import type { CommentArray } from 'comment-json'

export type KeybindingsContent = CommentArray<Record<string, string>> | null | undefined
