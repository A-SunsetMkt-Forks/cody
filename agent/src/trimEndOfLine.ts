export function trimEndOfLine(text: string | undefined): string {
    if (text === undefined) {
        return ''
    }
    return text
        .trim()
        .split('\n')
        .map(line => line.trimEnd())
        .join('\n')
}
