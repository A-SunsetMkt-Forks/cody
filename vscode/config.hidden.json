{
  "type": "object",
  "title": "Cody Hidden",
  "properties": {}
}
