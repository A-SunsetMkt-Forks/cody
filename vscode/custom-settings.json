{
    "cody.autocomplete.advanced.provider": "unstable-codegen",
    "cody.autocomplete.advanced.serverEndpoint": "https://backend4.codegen.salesforceresearch.ai/complete_batch",
    "cody.autocomplete.enabled": true,
    "cody.autocomplete.advanced.embeddings": true,

    "cody.useContext": "embeddings",
    "cody.experimental.chatPredictions": false,
    "cody.experimental.guardrails": false,
    "cody.plugins.enabled": false,
    "cody.inlineChat.enabled": false
}
