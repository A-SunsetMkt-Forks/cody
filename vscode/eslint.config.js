// @ts-check

const tslint = require('typescript-eslint')

module.exports = {
    languageOptions: {
        parser: tslint.parser,
        parserOptions: {
            project: ['tsconfig.json'],
        },
    },
    files: ['src/**/*.ts', 'src/**/*.tsx'],
    ignores: ['**/tree-sitter/**'],
    plugins: {
        '@typescript-eslint': tslint.plugin,
    },
    rules: {
        '@typescript-eslint/strict-boolean-expressions': ['error', {
            allowString: false,
            allowNumber: true,
            allowNullableObject: true,
            allowNullableBoolean: true,
            allowNullableString: true,
            allowNullableNumber: true,
            allowNullableEnum: true,
            allowAny: true,
            allowRuleToRunWithoutStrictNullChecksIKnowWhatIAmDoing: true,
        }],
    },
}
