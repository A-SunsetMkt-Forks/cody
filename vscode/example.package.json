"openctx.providers": {
          "type": "object",
          "markdownDescription": "OpenCtx providers configuration.",
          "default": {},
          "scope": "resource",
          "properties": {
            "A": {
              "type": "string"
            },
            "B": {
              "type": "string"
            }
          }
        },
