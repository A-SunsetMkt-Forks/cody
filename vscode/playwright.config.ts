import { defineConfig } from '@playwright/test'

export default defineConfig({
    workers: 1,
    // Give failing tests a second chance
    retries: 0, // TODO(sqs): back to 2
    testDir: 'test/e2e',
    expect: {
        timeout: 2000,
    },
})
