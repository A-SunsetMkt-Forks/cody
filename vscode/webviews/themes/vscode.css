body.vscode-dark {
    background: transparent; /* in vscode we should let the color of the underlying UI set the background of the sidebar*/
    /* Override VS Code Webview Toolkit elements */
    --border-width: none;
    color:var(--text);

    --font: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen', 'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
    --font-editor: 'Monaco', 'Consolas', monospace;
    --font-size: 13px;

    --vscode-font-family: var(--font);

    /* PRIMITIVES FROM EDITOR */
    --editor-background: var(--vscode-panel-background);
    --editor-border: var(--vscode-panel-border);
    --editor-accent: var(--vscode-button-background);
    --editor-text:var(--vscode-foreground);

    /* Primary Button */
    --button-primary-text: var(--vscode-button-foreground);
    --button-primary-background: var(--editor-accent);
    --button-primary-background-hover: var(--vscode-button-hoverBackground);
    --button-primary-border: var(--vscode-button-border);
    --button-primary-border-hover: var(--vscode-button-border);

    /* Backgrounds */
    --background-01: var(--editor-background);  /* Main page background */
    --background-02: color-mix(in srgb, var(--background-01), black 5%);  /* Slightly darker */
    --background-03: color-mix(in srgb, var(--background-01), black 10%);  /* Even darker */

    /* Text */
    --text: var(--editor-text);
    --text-subtle: color-mix(in srgb, var(--text) 70%, transparent 30%);  /* Slightly transparent */
    --text-muted: color-mix(in srgb, var(--text) 50%, transparent 50%);  /* More transparent */
    --text-primary: var(--editor-text);
    --text-primary-subtle: color-mix(in srgb, var(--text-primary)  70%, transparent 30%);  /* Subtle primary */
    --text-primary-muted: color-mix(in srgb, var(--text-primary)  50%, transparent 50%);  /* Muted primary */
    --text-disabled: color-mix(in srgb, var(--text-primary) 30%, transparent 70%);  /* Highly transparent */

    /* Links */
    --text-link-color: var(--editor-accent);
    --text-link-hover-color: #ff4c00;
    --text-white: #FFFFFF;

    /* Borders */
    --border: var(--editor-border);
    --border-subtle: color-mix(in srgb, var(--border) 70%, transparent 30%);  /* Subtle border */
    --border-disabled: color-mix(in srgb, var(--border) 30%, transparent 70%);  /* Disabled border */
    --border-active: color-mix(in srgb, var(--editor-accent), transparent 30%);  /* Active border with hint of blue */
    --border-hover: color-mix(in srgb, var(--border) 75%, white 25%);  /* Hover border with more vibrant blue */

    /* Highlight */
    --highlight-background: color-mix(in srgb, var(--editor-accent), transparent 40%);
    --highlight-text: color-mix(in srgb, var(--text), white 90%);
    --highlight-border: color-mix(in srgb, var(--editor-accent) 50%, white 20%);

    /* Links */
    --link-color: var(--editor-accent);
    --link-hover-color: color-mix(in srgb, var(--editor-accent) 75%, white 25%);
}

body.vscode-light {
    background:pink !important;
}
