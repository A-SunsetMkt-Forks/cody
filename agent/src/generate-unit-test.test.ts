import fs from 'node:fs'
import path from 'node:path'
import { afterAll, beforeAll, describe, expect, it } from 'vitest'
import * as vscode from 'vscode'
import type { ReplaceTextEdit } from '../../vscode/src/jsonrpc/agent-protocol'
import { TESTING_CREDENTIALS } from '../../vscode/src/testutils/testing-credentials'
import { TestClient } from './TestClient'
import { TestWorkspace } from './TestWorkspace'
import { trimEndOfLine } from './trimEndOfLine'

describe('Generate Unit Test', () => {
    const workspace = new TestWorkspace(path.join(__dirname, '__tests__', 'generate-unit-test'))
    const client = TestClient.create({
        workspaceRootUri: workspace.rootUri,
        name: 'generate-unit-test',
        credentials: TESTING_CREDENTIALS.dotcom,
    })

    beforeAll(async () => {
        await workspace.beforeAll()
        await client.beforeAll()
    })

    afterAll(async () => {
        await workspace.afterAll()
        await client.afterAll()
    })

    it('editCommands/test', async () => {
        const uri = workspace.file('src', 'trickyLogic.ts')
        await client.openFile(uri)
        const id = await client.request('editCommands/test', null)
        await client.taskHasReachedAppliedPhase(id)

        const untitledDocuments = client.workspace
            .allUris()
            .filter(uri => vscode.Uri.parse(uri).scheme === 'untitled')
        expect(untitledDocuments).toHaveLength(2)
        const untitledDocument = untitledDocuments.find(d => d.endsWith('trickyLogic.spec.ts'))
        expect(untitledDocument).toBeDefined()
        const testDocument = client.workspace.getDocument(vscode.Uri.parse(untitledDocument ?? ''))
        expect(trimEndOfLine(testDocument?.getText())).toMatchSnapshot()
        expect(client.textDocumentEditParams).toHaveLength(1)
        for (const editParam of client.textDocumentEditParams) {
            for (const edit of editParam.edits) {
                const range = (edit as ReplaceTextEdit).range
                expect(range.start.line).toBeGreaterThanOrEqual(0)
                expect(range.end.line).toBeGreaterThanOrEqual(0)
            }
        }
    }, 30_000)

    it('respects existing content by inserting on a blank line', async () => {
        const testUri = workspace.file('src', 'existing', 'body.spec.ts')
        // We use fs here so we don't open the file yet. This is so we can catch
        // caching issues
        const testBeforeContent = await fs.promises.readFile(testUri.fsPath, 'utf-8')

        const uri = workspace.file('src', 'existing', 'body.ts')
        await client.openFile(uri)
        const id = await client.request('editCommands/test', null)
        await client.taskHasReachedAppliedPhase(id)

        const testFile = client.workspace.getDocument(testUri)

        const testAfterContent = testFile?.getText()

        expect(testAfterContent?.startsWith(testBeforeContent)).toBeTruthy()
        expect(testAfterContent).not.toEqual(testBeforeContent)
    })

    it('handles content changes outside the editor', async () => {
        const testUri = workspace.file('src', 'existing', 'body.spec.ts')
        const testBeforeContent = await fs.promises.readFile(testUri.fsPath, 'utf-8')

        // we open and "close" the file to hopefully cache the content.
        // technically I would like to just close the editor but there does not
        // seem to be a function for that yet
        await client.openFile(testUri)
        // await client.workspace.reset()

        // after which we modify the file outside the editor. This could be like a `git checkout` command
        await fs.promises.writeFile(testUri.fsPath, 'blank content\n')

        // now we generate a test to see if the testfile content is stale
        const uri = workspace.file('src', 'existing', 'body.ts')

        // This call should not be neccesary:
        // await client.openFile(testUri)
        await client.openFile(uri)
        const id = await client.request('editCommands/test', null)
        await client.taskHasReachedAppliedPhase(id)

        const testFile = client.workspace.getDocument(testUri)

        const testAfterContent = testFile?.getText()

        // the test file should now no longer start with the original content
        // but with "blank content"
        expect(testAfterContent?.startsWith(testBeforeContent)).toBeFalsy()
    })
})
