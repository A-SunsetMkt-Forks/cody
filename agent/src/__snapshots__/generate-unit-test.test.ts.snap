// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Generate Unit Test > editCommands/test 1`] = `
"
import { trickyLogic } from './trickyLogic'

describe('trickyLogic', () => {
    it('should return 1 when a is 0', () => {
        const result = trickyLogic(0, 5)
        expect(result).toBe(1)
    })

    it('should return 1 when b is 2', () => {
        const result = trickyLogic(3, 2)
        expect(result).toBe(1)
    })

    it('should return a - b when a is not 0 and b is not 2', () => {
        const result = trickyLogic(5, 3)
        expect(result).toBe(2)
    })

    it('should handle negative numbers', () => {
        const result = trickyLogic(-2, -1)
        expect(result).toBe(-1)
    })
})
"
`;
